src/utilities/include/par.h

```diff
@@ -529,11 +526,13 @@ template <typename Iter, typename P,
 Iter remove_if(ExecutionPolicy policy, Iter first, Iter last, P pred) {
 #if MANIFOLD_PAR == 'T'
   if (policy == ExecutionPolicy::Par) {
-    std::vector<T> tmp(std::distance(first, last));
-    auto back = copy_if(policy, first, last, tmp.begin(),
-                        [&](T v) { return !pred(v); });
-    copy(policy, tmp.begin(), back, first);
-    return first + std::distance(tmp.begin(), back);
+    T *tmp = new T[std::distance(first, last)];
+    auto back =
+        copy_if(policy, first, last, tmp, [&](T v) { return !pred(v); });
+    copy(policy, tmp, back, first);
+    auto d = std::distance(tmp, back);
+    delete[] tmp;
```

Comment:
Do we have any other usages of `new` or `delete` in this code base? Why is this better than a vector?

Reply:
The idea is to avoid initializing the memory. But it is probably premature optimization...

Reply:
nvm, I did some benchmark, this is not premature. It does matter for some of the benchmarks.

Reply:
I sort of thought this was a feature of `Vec` - that `Vec<int> a(10)` was uninitialized and `Vec<int> a(10, 0)` was initialized to 0. Should it be?

Reply:
Yes, but I would rather avoid introducing dependency to `Vec` here because `Vec` also use some parallel algorithms (copy and fill).